File: dota-zone/dotazone/src/main/java/br/com/dotazone/view/activity/MainActivity.java
package br.com.dotazone.view.activity;

import android.os.Bundle;
import androidx.drawerlayout.widget.DrawerLayout;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import com.google.android.gms.analytics.GoogleAnalytics;

import br.com.dotazone.DotaZoneApplication;
import br.com.dotazone.DotazoneMenu;
import br.com.dotazone.R;
import br.com.dotazone.view.fragment.FeedNewsFragment;

public class MainActivity extends BaseActivity implements OnClickListener {

    public DotazoneMenu mMenu;
    private DrawerLayout mDrawerLayout;
    private RelativeLayout mDrawerList;
    private ImageView mImageLogo;
    private ImageView mImageMenu;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ((DotaZoneApplication) getApplication()).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER);

        initComponents();
        FeedNewsFragment fragment1 = FeedNewsFragment.Companion.newInstance();
        if (getIntent() != null && getIntent().getExtras() != null
                && getIntent().getExtras().getInt(MainActivity.class.getName()) == DotazoneMenu.MENU_NEWS) {

            getSupportFragmentManager().beginTransaction().replace(R.id.main_activity_content, fragment1).commit();

        } else {
            getSupportFragmentManager().beginTransaction().replace(R.id.main_activity_content, fragment1).commit();
            mMenu.checkNewsMenu();
        }
        // Get a Tracker (should auto-report)
        ((DotaZoneApplication) getApplication()).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER);


        if (getIntent() != null && getIntent().getExtras() != null
                && getIntent().getExtras().getInt(MainActivity.class.getName()) == 100) {


        }


    }

    @Override
    protected void onResume() {
        super.onResume();

        if (getIntent() != null && getIntent().getExtras() != null
                && getIntent().getExtras().getInt(MainActivity.class.getName()) == DotazoneMenu.MENU_NEWS) {

            mMenu.checkNewsMenu();
        }
    }

    @Override
    protected void onStart() {
        super.onStart();

        GoogleAnalytics.getInstance(this).reportActivityStart(this);
    }

    @Override
    protected void onStop() {
        super.onStop();

        // Stop the analytics tracking
        GoogleAnalytics.getInstance(this).reportActivityStop(this);
    }

    @Override
    public void initComponents() {

        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
        mDrawerList = (RelativeLayout) findViewById(R.id.list_slidermenu);
        mMenu = new DotazoneMenu(this, mDrawerLayout, mDrawerList);
        mImageLogo = (ImageView) findViewById(R.id.header_imageLogo);
        mImageMenu = (ImageView) findViewById(R.id.header_imageMenu);
        mImageLogo.setOnClickListener(this);
        mImageMenu.setOnClickListener(this);
    }

    @Override
    public void setActionErrorOk() {
        //finish();
    }

    @Override
    public void setActionErrorCancel() {
        // TODO Auto-generated method stub

    }

    @Override
    public void onClick(View v) {

        mDrawerLayout.openDrawer(mDrawerList);

    }

    @Override
    public void finish() {
        super.finish();
        mMenu.defaultMenu();
    }
}

File: dota-zone/app/src/main/java/com/br/dotazone/view/activity/MainActivity.kt
package com.br.dotazone.view.activity

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout
import androidx.drawerlayout.widget.DrawerLayout
import com.br.dotazone.DotaZoneApplication
import com.br.dotazone.R
import com.br.dotazone.view.fragment.FeedNewsFragment.Companion.newInstance
import com.com.dotazone.DotazoneMenu
import com.google.android.gms.analytics.GoogleAnalytics


class MainActivity : BaseActivity(), View.OnClickListener {
	var mMenu: DotazoneMenu? = null
	private var mDrawerLayout: DrawerLayout? = null
	private var mDrawerList: RelativeLayout? = null
	private var mImageLogo: ImageView? = null
	private var mImageMenu: ImageView? = null
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)
		(application as DotaZoneApplication).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER)
		initComponents()
		val fragment1 = newInstance()
		if (intent != null && intent.extras != null && intent.extras!!.getInt(MainActivity::class.java.name) == DotazoneMenu.MENU_NEWS) {
			supportFragmentManager.beginTransaction().replace(R.id.main_activity_content, fragment1).commit()
		} else {
			supportFragmentManager.beginTransaction().replace(R.id.main_activity_content, fragment1).commit()
			mMenu!!.checkNewsMenu()
		}
		// Get a Tracker (should auto-report)
		(application as DotaZoneApplication).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER)
		if (intent != null && intent.extras != null && intent.extras!!.getInt(MainActivity::class.java.name) == 100) {
		}
	}

	override fun onResume() {
		super.onResume()
		if (intent != null && intent.extras != null && intent.extras!!.getInt(MainActivity::class.java.name) == DotazoneMenu.MENU_NEWS) {
			mMenu!!.checkNewsMenu()
		}
	}

	override fun onStart() {
		super.onStart()
		GoogleAnalytics.getInstance(this).reportActivityStart(this)
	}

	override fun onStop() {
		super.onStop()

		// Stop the analytics tracking
		GoogleAnalytics.getInstance(this).reportActivityStop(this)
	}

	override fun initComponents() {
		mDrawerLayout = findViewById<View>(R.id.drawer_layout) as DrawerLayout
		mDrawerList = findViewById<View>(R.id.listSliderMenu) as RelativeLayout
		mMenu = DotazoneMenu(this, mDrawerLayout, mDrawerList)
		mImageLogo = findViewById<View>(R.id.header_imageLogo) as ImageView
		mImageMenu = findViewById<View>(R.id.header_imageMenu) as ImageView
		mImageLogo!!.setOnClickListener(this)
		mImageMenu!!.setOnClickListener(this)
	}

	override fun setActionErrorOk() {
		//finish();
	}

	override fun setActionErrorCancel() {
		// TODO Auto-generated method stub
	}

	override fun onClick(v: View) {
		mDrawerLayout!!.openDrawer(mDrawerList!!)
	}

	override fun finish() {
		super.finish()
		mMenu!!.defaultMenu()
	}
}
